### PR TITLE
fixed linker being slow on OSX

### PR DESCRIPTION
**Pull request by RevoluPowered**
linker is slow on OSX when using debug builds, these options improve it drastically (80s to 20s)

### CODE REVIEW

**Comment by naithar** [comment]
Would the same change improve iOS builds too?

File: platform/osx/detect.py [hunk 1]
```diff
@@ -76,6 +76,7 @@ def configure(env):
     elif env["target"] == "debug":
         env.Prepend(CCFLAGS=["-g3"])
         env.Prepend(CPPDEFINES=["DEBUG_ENABLED"])
+        env.Prepend(LINKFLAGS=["-Xlinker", "-no_deduplicate"])
```
Comment (akien-mga) [inline platform/osx/detect.py:79]: Doesn't it result in bigger, and possibly worse optimized binaries?

Comment (RevoluPowered) [reply platform/osx/detect.py:79]: For debug this doesn't matter IMHO, it sets these by default in xcode too normally for debug:
- Bin size: 198 MB with this setting on
- Bin size: 190 MB with this setting off

Before:
scons -j16  58.31s user 3.16s system 111% cpu 55.303 total

After:
scons -j16  20.81s user 3.21s system 141% cpu 17.007 total

**Review by bruvzg — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.3.

**Pull request merged**